engine/scheduler.cpp

```diff
@@ -206,6 +212,18 @@ struct test_exec_data : public exec_data {
     /// denote that no further attempts shall be made at cleaning this up.
     bool needs_cleanup;
 
+    /// Whether this test case still needs to have its execenv cleanup executed.
+    ///
+    /// This is set externally when the cleanup routine is actually invoked to
+    /// denote that no further attempts shall be made at cleaning this up.
+    bool needs_execenv_cleanup;
+
+    /// Original PID of the test case subprocess.
+    ///
+    /// This is used for the cleanup upon termination by a signal, to reap the
+    /// leftovers and form missing exit_handle.
+    int pid;
```

Comment:
_(picking a line)_
`pid_t` is the POSIXly correct type.
```suggestion
    pid_t pid;
```

Reply:
Fixed.